c++/src/capnp/reconnect.h

```diff
@@ -52,13 +52,26 @@ auto autoReconnect(ConnectFunc&& connect);
 // to avoid an infinite loop in the case that the DISCONNECTED exception actually represents a
 // permanent problem. Consider using `kj::retryOnDisconnect()` to implement this behavior.
 
+template <typename ConnectFunc>
+auto lazyAutoReconnect(ConnectFunc&& connect);
+// The same as autoReconnect, but doesn't call the provided connect function until the first
+// time the capability is used. Note that only the initial connection is lazy -- upon
+// disconnected errors this will still reconnect eagerly.
+
 // =======================================================================================
 // inline implementation details
 
-Capability::Client autoReconnect(kj::Function<Capability::Client()> connect);
+Capability::Client autoReconnect(kj::Function<Capability::Client()> connect, bool lazy = false);
 template <typename ConnectFunc>
 auto autoReconnect(ConnectFunc&& connect) {
-  return autoReconnect(kj::Function<Capability::Client()>(kj::fwd<ConnectFunc>(connect)))
+  return autoReconnect(kj::Function<Capability::Client()>(kj::fwd<ConnectFunc>(connect)), false)
+      .castAs<FromClient<kj::Decay<decltype(connect())>>>();
+}
+
+Capability::Client lazyAutoReconnect(kj::Function<Capability::Client()> connect);
```

Comment:
Were you intending to define this in the `.c++` file? It looks like you ended up adding a `bool lazy` parameter to `autoReconnect()` instead.

I think I'd prefer having the two named declarations rather than the boolean parameter visible in the header.

Reply:
Looks like I did a little of column A and a little of column B then forgot to clean one of them up. Switched over to the separate declarations.